src/core/device_manager.py

```
@@ -441,288 +572,353 @@ def _get_unauthorized_device_info(self, serial: str, metadata: Dict[str, str] =
             self.logger.error(f"Error checking FRP status for {serial}: {e}")
-        
-        return 'unknown'
-    
+
+        return "unknown"
+
     def _get_device_imei(self, serial: str) -> str:
         """Get device IMEI (requires appropriate permissions)"""
         try:
             # Try service call method
             result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)